src/backend/services/metrics.py

```diff
@@ -372,9 +372,9 @@ def initialize_sdk_metrics_data(
         MetricsData(
             id=str(uuid.uuid4()),
             message_type=message_type,
-            trace_id=kwargs.pop("trace_id", None),
-            user_id=kwargs.pop("user_id", None),
-            assistant_id=kwargs.pop("agent_id", None),
+            trace_id=kwargs.get("trace_id", None),
```

Comment:
i like that, probably would be easier to debug if its "unknown" since we can ref back to here.

Reply:
unrelated, but why do we pop the kwargs? would that not mutate for future uses of it leading to unexpected behaviour?

re: "unknown", would we still have a way of catching schema errors when sending through metrics? would it simply being a string cause silent errors?

Reply:
lets talk about this as a follow up. i will change it back to None and merge this PR. since this change isnt really related to the PR but rather just a piggyback